src/helpers/dimensions.ts

```diff
@@ -0,0 +1,166 @@
+    }, {
+        itemIdentifiers: measuresItemIdentifiers
+    }];
+}
+
+function getPieDimensions(mdObject: VisualizationObject.IVisualizationObjectContent): AFM.IDimension[] {
+    const view = mdObject.buckets.find(bucket => bucket.localIdentifier === VIEW);
+
+    if (view && view.items.length) {
+        return [
+            {
+                itemIdentifiers: [MEASUREGROUP]
+            },
+            {
+                itemIdentifiers: view.items.map((attribute: VisualizationObject.IVisualizationAttribute) =>
+                    attribute.visualizationAttribute.localIdentifier)
+            }
+        ];
+    }
+
+    return [
+        {
+            itemIdentifiers: []
+        },
+        {
+            itemIdentifiers: [MEASUREGROUP]
+        }
+    ];
+}
+
+function getBarDimensions(mdObject: VisualizationObject.IVisualizationObjectContent): AFM.IDimension[] {
+    const view: VisualizationObject.IBucket = mdObject.buckets
+            .find(bucket => bucket.localIdentifier === VIEW);
+
+    const stack: VisualizationObject.IBucket = mdObject.buckets
+        .find(bucket => bucket.localIdentifier === STACK);
+
+    const hasNoStacks = !stack || !stack.items || stack.items.length === 0;
+
+    if (hasNoStacks) {
+        return [
+            {
+                itemIdentifiers: [MEASUREGROUP]
+            },
+            {
+                itemIdentifiers: (view && view.items || [])
+                    .map((v: VisualizationObject.IVisualizationAttribute) =>
+                        v.visualizationAttribute.localIdentifier)
+            }
+        ];
+    }
+
+    return [
+        {
+            itemIdentifiers: (stack && stack.items || [])
+                .map((s: VisualizationObject.IVisualizationAttribute) => s.visualizationAttribute.localIdentifier)
+        },
+        {
+            itemIdentifiers: ((view && view.items || [])
+                .map((v: VisualizationObject.IVisualizationAttribute) => v.visualizationAttribute.localIdentifier))
+                .concat([MEASUREGROUP])
+        }
+    ];
+}
+
+function getLineDimensions(mdObject: VisualizationObject.IVisualizationObjectContent): AFM.IDimension[] {
+    const trend: VisualizationObject.IBucket = mdObject.buckets
+        .find(bucket => bucket.localIdentifier === TREND);
+
+    const segment: VisualizationObject.IBucket = mdObject.buckets
+        .find(bucket => bucket.localIdentifier === SEGMENT);
+    const hasNoSegments = !segment || !segment.items || segment.items.length === 0;
+
+    if (hasNoSegments) {
+        return [
+            {
+                itemIdentifiers: [MEASUREGROUP]
+            },
+            {
+                itemIdentifiers: (trend && trend.items || [])
+                    .map((t: VisualizationObject.IVisualizationAttribute) =>
+                        t.visualizationAttribute.localIdentifier)
+            }
+        ];
+    }
+
+    return [
+        {
+            itemIdentifiers: (segment && segment.items || [])
```

Comment:
few line above there is `hasNoSegment` variable

Reply:
always better to be defensive. Condition can be changed, or this code can be moved outside of else statement.